src/MIDebugEngine/Engine.Impl/Breakpoints.cs

```diff
@@ -329,6 +377,12 @@ internal BoundBreakpoint(PendingBreakpoint parent, ulong addr, /*optional*/ Tupl
             }
         }
 
+        internal BoundBreakpoint(PendingBreakpoint parent, ulong addr, uint size)
+        {
+            Addr = addr;
```

Comment:
Aren't we using Addr for two kind of unrelated purposes here -- address of the instruction where the breakpoint is set vs. address of the data being watched.